### PR TITLE
[Docs] Replace markdownlint with mdformat and configure myst-parser

### PR DESCRIPTION
**Pull request by RunningLeon**
## Motivation

Following:
Replace markdownlint: https://github.com/open-mmlab/mmcv/pull/1936 
Configure Myst-parser to parse anchor tag:  https://github.com/open-mmlab/mmocr/pull/1012

## Modification

md docs

## BC-breaking (Optional)
None

### CODE REVIEW

**Comment by lvhan028** [comment]
Let's deal with it when dev-0.5.0 is ready. Otherwise, lots of conflicts will drive us mad

File: mmdeploy/apis/pytorch2onnx.py [hunk 1]
```diff
@@ -27,12 +27,12 @@ def torch2onnx(img: Any,
         >>> img = 'demo.jpg'
         >>> work_dir = 'work_dir'
         >>> save_file = 'fcos.onnx'
-        >>> deploy_cfg = ('configs/mmdet/detection/'
-                          'detection_onnxruntime_dynamic.py')
-        >>> model_cfg = ('mmdetection/configs/fcos/'
-                         'fcos_r50_caffe_fpn_gn-head_1x_coco.py')
-        >>> model_checkpoint = ('checkpoints/'
-                                'fcos_r50_caffe_fpn_gn-head_1x_coco-821213aa.pth')
+        >>> deploy_cfg = 'configs/mmdet/detection/' \
```
Comment (lvhan028) [inline mmdeploy/apis/pytorch2onnx.py:30]: Is this line broken by pre-commit automatically? '\\' makes API docs not pretty in readthedocs.  We used to fix it in PR #495

File: mmdeploy/core/optimizers/function_marker.py [hunk 2]
```diff
@@ -154,7 +154,7 @@ def impl(ys, prefix, level):
             if ys not in visit:
                 visit.add(ys)
                 root = ctx.names[ctx.index]
-                name = '.'.join(str(x) for x in (root, *prefix))
+                name = '/'.join(str(x) for x in (root, *prefix))
```
Comment (lvhan028) [inline mmdeploy/core/optimizers/function_marker.py:157]: I think it has been fixed in another PR. May rebase master

File: mmdeploy/codebase/mmdet/core/post_processing/bbox_nms.py [hunk 3]
```diff
@@ -201,7 +201,7 @@ def multiclass_nms__default(ctx,
     """
     deploy_cfg = ctx.cfg
     batch_size = boxes.size(0)
-    if not is_dynamic_batch(deploy_cfg) and batch_size == 1:
+    if not is_dynamic_batch(deploy_cfg) and batch_size != 1:
```
Comment (lvhan028) [inline mmdeploy/codebase/mmdet/core/post_processing/bbox_nms.py:204]: I think it has been fixed in another PR. May rebase master

File: README_zh-CN.md [hunk 4]
```diff
@@ -51,38 +52,41 @@ MMDeploy 是 [OpenMMLab](https://openmmlab.com/) 模型部署工具箱，**为
 - [mmrotate](docs/en/04-supported-codebases/mmrotate.md)
 
 ### 支持多种推理后端
+
 | ONNX Runtime | TensorRT | ppl.nn | ncnn | OpenVINO | more                                              |
 | ------------ | -------- | ------ | ---- | -------- | ------------------------------------------------- |
-| ✔️            | ✔️        | ✔️      | ✔️    | ✔️        | [benchmark](docs/zh_cn/03-benchmark/benchmark.md) |
+| ✔️           | ✔️       | ✔️     | ✔️   | ✔️       | [benchmark](docs/zh_cn/03-benchmark/benchmark.md) |
```
Comment (lvhan028) [inline README_zh-CN.md:58]: Hi, @zhouzaida `mdformat` adjusts this table by mistake.

File: mmdeploy/apis/inference.py [hunk 5]
```diff
@@ -16,15 +16,15 @@ def inference_model(model_cfg: Union[str, mmcv.Config],
 
     Examples:
         >>> from mmdeploy.apis import inference_model
-        >>> model_cfg = ('mmdetection/configs/fcos/'
-                         'fcos_r50_caffe_fpn_gn-head_1x_coco.py')
-        >>> deploy_cfg = ('configs/mmdet/detection/'
-                          'detection_onnxruntime_dynamic.py')
+        >>> model_cfg = 'mmdetection/configs/fcos/' \
```
Comment (lvhan028) [inline mmdeploy/apis/inference.py:19]: Is this line broken by pre-commit automatically? '\' makes API docs not pretty in readthedocs. We used to fix it in PR #495

Comment (RunningLeon) [reply mmdeploy/apis/inference.py:19]: I'll update it. maybe there's a mistake when revert merge commit.

File: mmdeploy/core/optimizers/function_marker.py (hunk 2 above)
Comment (RunningLeon) [reply mmdeploy/core/optimizers/function_marker.py:157]: Yes. Before there are a lot of conflicts. To make it clean, I reverted some commits. Maybe something wrong when revert the merge commit. `rebase` to master not work in this case.

**Comment by RunningLeon** [comment]
> How about we drop this PR and apply "markdownlint: [open-mmlab/mmcv#1936](https://github.com/open-mmlab/mmcv/pull/1936)" and "[open-mmlab/mmocr#1012](https://github.com/open-mmlab/mmocr/pull/1012)" to the latest mmdeploy directly?

OK

**Comment by RunningLeon** [comment]
move to https://github.com/open-mmlab/mmdeploy/pull/610

**Pull request closed without merging**